services/QuillLMS/app/services/skills_report/user_responses_processor.rb

```diff
@@ -21,11 +21,16 @@ def run
       Result.new(name:, ema:, ema_previous:, trend:, exposure:, unique_answered:, replays_answered:, total_answered:, replay_rate:, last_practiced:, scores:, timespent:)
     end
 
+    private def ema_trends = @ema_trends ||= EmaTrendCalculator.run(scores:)
+    private def latest_response = @latest_response ||= non_empty_responses.first
     private def non_empty_responses = @non_empty_responses ||= responses.filter { |r| r.activity_session_id.present? && r.question_uid.present? }
-    private def unique_answered = @unique_answered ||= non_empty_responses.uniq { |r| r.question_uid }.count
-    private def total_answered = @total_answered ||= non_empty_responses.count
+    private def questions_to_score = non_empty_responses.first(EmaTrendCalculator::MAX_RECORDS_NEEDED)
+    private def replay_rate = (replays_answered / [total_answered.to_f, 1].max) * 100
     private def replays_answered = total_answered - unique_answered
-    private def latest_response = @latest_response ||= non_empty_responses.first
+    private def scores = @scores ||= questions_to_score.map { |response| QuestionResponseScorer.run(response:, question: questions_hash[response.question_uid].first) }
+    private def total_answered = @total_answered ||= non_empty_responses.count
+    private def unique_answered = @unique_answered ||= non_empty_responses.uniq { |r| r.question_uid }.count
```

Comment:
This is a re-organization of one-line method definitions, none of this code is actually changed.